modules/govuk_jenkins/templates/jobs/deploy_emergency_banner.yaml.erb

```diff
@@ -25,7 +25,59 @@
           LINK=$(escapeRakeArg $LINK)
           LINK_TEXT=$(escapeRakeArg $LINK_TEXT)
 
+          echo "Deploying emergency banner"
           ssh deploy@$(govuk_node_list -c frontend --single-node) "cd /var/apps/static && govuk_setenv static bundle exec rake \"emergency_banner:deploy[$CAMPAIGN_CLASS,$HEADING,$SHORT_DESCRIPTION,$LINK,$LINK_TEXT]\""
+
+          echo "Clearing template cache.."
+          for node in `govuk_node_list frontend`; do
+            echo "-- $node"
+            ssh deploy@$node "sudo rm /var/apps/static/public/templates/{wrapper,homepage,header_footer_only,core_layout}.html.erb"
+            ssh deploy@$node "sudo rm -rf /var/apps/{,government-}frontend/tmp/cache/*"
+          done
+          for node in `govuk_node_list calculators_frontend`; do
+            echo "-- $node"
+            ssh deploy@$node "sudo rm -rf /var/apps/finder-frontend/tmp/cache/*"
+          done
+
+          echo "Restarting memcached.."
+          for node in `govuk_node_list "frontend,whitehall_frontend"`; do
+            echo "-- $node"
+            ssh deploy@$node "sudo service memcached restart"
+          done
+
+          echo "Clearing varnish cache.."
+          for node in `govuk_node_list cache`; do
+            echo "-- $node"
+            ssh deploy@$node "sudo varnishadm 'ban req.url ~ .'"
+          done
+
+          <% if @clear_cdn_cache %>
```

Comment:
Something that makes me a bit nervous here - but not something where I know a particularly good solution - is that I don't think we'll ever get to test this code and would only find out if it's good at the point of doing this for real. 

Admittedly this is more a problem with our infrastructure than the script. 

Best I can think of for this is to have some sort of help message alert before it of a link for if it fails. 

Sidenote: I wonder how this small set of paths got defined - they seem quite abstract.

Reply:
Perhaps if we take @thomasleese's suggestion to the extreme we could extract CDN clearing into its own job which we could test during a quiet period.

As for the paths, those are (were?) the top 10 most visited paths according to the fabric scripts.  I'll put a comment in to that effect and see if they're still current.